File: server/PlayItLoud.API/Models/Album.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlayItLoud.API.Models
{
    [Table("Album")]
    public class Album
    {
        public int Id { get; private set; }
        [Required]
        [StringLength(1000)]
        public string Name { get; set; }
        public string Cover { get; set; } = "/images/default.png";
        public DateTime? ReleaseDate { get; set; }
        public string? Description { get; set; }
        public string? Label { get; set; }
        public ICollection<Artist> Artists { get; set; } = new List<Artist>();
        public ICollection<Genre> Genres { get; set; } = new List<Genre>();
        public ICollection<Track> Tracks { get; set; } = new List<Track>();
    }
}

File: client/src/app/albums/models/album.ts
import { Artist } from "../../artists/models/artist";
import { Genre } from "../../genres/models/genre";
import { Track, TrackDTO } from "./track";

export interface Album {
    id: number;
    name: string;
    cover: string;
    releaseDate?: Date;
    description?: string;
    label?: string;
    artists: Artist[];
    genres: Genre[];
    tracks: Track[];
  }

  export interface AlbumDTO {
    name: string;
    releaseDate?: Date;
    description?: string;
    label?: string;
    artistIds: number[];
    genreIds: number[];
    tracks: TrackDTO[];
  }